### PR TITLE
Fix static StorageMap

### LINKED ISSUE #310: StorageMap can't be written in static variables
I found that if StorageMap is written in a global variables, the execution will report an error. 
If the StorageMap is written in a variable in the method, it is executed correctly.

``` c#
[Features(ContractFeatures.HasStorage)]
public class Contract1 : SmartContract
{
    private static readonly StorageMap Data = Storage.CurrentContext.CreateMap("data");

    public static void Put(string message)
    {
        Data.Put(message, Blockchain.GetHeight());
    }

    public static BigInteger Get(string msg)
    {
        return Data.Get(msg)?.ToBigInteger() ?? 0;
    }

    public static void Put2(string message)
    {
        var Data2 = Storage.CurrentContext.CreateMap("data");
        Data2.Put(message, Blockchain.GetHeight());
    }

    public static BigInteger Get2(string msg)
    {
        var Data2 = Storage.CurrentContext.CreateMap("data");
        return Data2.Get(msg)?.ToBigInteger() ?? 0;
    }
}
```

The `put` and `get` methods will fail to execute, but the `put2` and `get2` methods will execute successfully.

**This bug will appear in both Neo2 and Neo3**

The picture below is the execution result:

image

StorageMap 如果写在全局变量（静态变量）中，会执行出错，但写在方法里面就能正常执行，从Neo2到现在一直存在的Bug。为此我还补充了文档。但我觉得有坑的话应该先把坑填上，而不是叫所有人都绕过去走。

image

### PR DESCRIPTION
**Pull request by ShawnYun**
Close #310 
If there is a syscall in cctor, we should add it into staticfieldsCctor directly.

### CODE REVIEW

File: src/Neo.Compiler.MSIL/MSIL/CctorSubVM.cs [hunk 1]
```diff
@@ -207,6 +210,12 @@ public static bool Parse(ILMethod from, NeoModule to)
                                             calcStack.Push(n);
                                         }
                                     }
+                                    if (attr.AttributeType.FullName == "Neo.SmartContract.Framework.SyscallAttribute")
+                                    {
+                                        haveSysCall = true;
```
Comment (Tommo-L) [inline src/Neo.Compiler.MSIL/MSIL/CctorSubVM.cs:215]:
```suggestion
                                        ret = true;
```
Add comments here

**Comment by Tommo-L** [comment]
@lightszero dalao, could have a check?

**Comment by superboyiii** [comment]
@ShawnYun According to the contract in #310, I can't compile it based on this PR.
```
D:\Master-smoking\2020-07-09\neo-devpack-dotnet\NeoContract\bin\Debug\netstandard2.1>neon -f "D:\Master-smoking\2020-07-09\neo-devpack-dotnet\NeoContract\bin\Debug\netstandard2.1\NeoContract.dll" -o
Neo.Compiler.MSIL console app v3.0.0.0
Convert Error:System.Exception: error:System.Void NeoContract.Contract1::Put(System.String)::IL_0001 Ldsfld (15)
 ---> System.Collections.Generic.KeyNotFoundException: The given key 'Neo.SmartContract.Framework.Services.Neo.StorageMap NeoContract.Contract1::Data' was not present in the dictionary.
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at Neo.Compiler.MSIL.ModuleConverter.ConvertCode(ILMethod method, OpCode src, NeoMethod to) in D:\Master-smoking\2020-07-09\neo-devpack-dotnet\src\Neo.Compiler.MSIL\MSIL\Converter.cs:line 908
   at Neo.Compiler.MSIL.ModuleConverter.FillMethod(ILMethod from, NeoMethod to, Boolean withReturn) in D:\Master-smoking\2020-07-09\neo-devpack-dotnet\src\Neo.Compiler.MSIL\MSIL\Converter.cs:line 265
   --- End of inner exception stack trace ---
   at Neo.Compiler.MSIL.ModuleConverter.FillMethod(ILMethod from, NeoMethod to, Boolean withReturn) in D:\Master-smoking\2020-07-09\neo-devpack-dotnet\src\Neo.Compiler.MSIL\MSIL\Converter.cs:line 269
   at Neo.Compiler.MSIL.ModuleConverter.Convert(ILModule _in, ConvOption option) in D:\Master-smoking\2020-07-09\neo-devpack-dotnet\src\Neo.Compiler.MSIL\MSIL\Converter.cs:line 95
   at Neo.Compiler.Program.Compile(Options options) in D:\Master-smoking\2020-07-09\neo-devpack-dotnet\src\Neo.Compiler.MSIL\Program.cs:line 163
```

**Comment by superboyiii** [comment]
Tested, it works.
```
neo> invoke 0x4d4c520e65b4ad21267ce26e63f3841f7d4a31fb put [{"type":"String","value":"erik"}]
Invoking script with: '0c046572696b11c00c037075740c14fb314a7d1f84f3636ee27c2621adb4650e524c4d41627d5b52'
VM State: HALT
Gas Consumed: 0.033917
Result Stack: [{"type":"Any"}]
Relay tx(no|yes): y
Signed and relayed transaction with hash=0x346f696140c296b6b177634d18837bd4a14052eb8127697e6f3dd8bfe28d28b2
neo> invoke 0x4d4c520e65b4ad21267ce26e63f3841f7d4a31fb get [{"type":"String","value":"data"}]
Invoking script with: '0c046461746111c00c036765740c14fb314a7d1f84f3636ee27c2621adb4650e524c4d41627d5b52'
VM State: HALT
Gas Consumed: 0.0339177
Result Stack: [{"type":"Integer","value":"0"}]
Relay tx(no|yes): n
neo> invoke 0x4d4c520e65b4ad21267ce26e63f3841f7d4a31fb get [{"type":"String","value":"erik"}]
Invoking script with: '0c046572696b11c00c036765740c14fb314a7d1f84f3636ee27c2621adb4650e524c4d41627d5b52'
VM State: HALT
Gas Consumed: 0.0347161
Result Stack: [{"type":"Integer","value":"2537"}]
Relay tx(no|yes): n
neo> invoke 0x4d4c520e65b4ad21267ce26e63f3841f7d4a31fb put2 [{"type":"String","value":"shaoyun"}]
Invoking script with: '0c077368616f79756e11c00c04707574320c14fb314a7d1f84f3636ee27c2621adb4650e524c4d41627d5b52'
VM State: HALT
Gas Consumed: 0.0435077
Result Stack: [{"type":"Any"}]
Relay tx(no|yes): y
Signed and relayed transaction with hash=0x7295411f3cfaac21b0c52df0483526dde29339d5fefa6d73f805e02a6ce3a7e2
neo> invoke 0x4d4c520e65b4ad21267ce26e63f3841f7d4a31fb get2 [{"type":"String","value":"shaoyun"}]
Invoking script with: '0c077368616f79756e11c00c04676574320c14fb314a7d1f84f3636ee27c2621adb4650e524c4d41627d5b52'
VM State: HALT
Gas Consumed: 0.0413068
Result Stack: [{"type":"Integer","value":"2611"}]
Relay tx(no|yes): n
```

**Comment by superboyiii** [comment]
@shargon Could you merge this?

**Comment by Tommo-L** [comment]
@shargon Merge?

**Pull request merged**